src/nupic/ntypes/BasicType.cpp

```diff
@@ -281,13 +281,14 @@ void BasicType::convertArray(void *ptr1, NTA_BasicType toType, const void *ptr2,
     case NTA_BasicType_Int16:
       switch (toType) {
       case NTA_BasicType_Byte:
-        cpyarray<Byte, Int16>(ptr1, ptr2, count, (Int16)std::numeric_limits<Byte>::min(), (Int16)std::numeric_limits<Byte>::max());
+        cpyarray<Byte, Int16>(ptr1, ptr2, count, static_cast<Int16>(std::numeric_limits<Byte>::min()), 
+                                                 static_cast<Int16>(std::numeric_limits<Byte>::max()));
         break;
       case NTA_BasicType_Int16:
         cpyarray<Int16, Int16>(ptr1, ptr2, count);
         break;
       case NTA_BasicType_UInt16:
-        cpyarray<UInt16, Int16>(ptr1, ptr2, count, 0, (UInt16)std::numeric_limits<Int16>::max());
+        cpyarray<UInt16, Int16>(ptr1, ptr2, count, 0, static_cast<Int16>(std::numeric_limits<Int16>::max()));
```

Comment:
is this correct? `UInt16 vs <Int16> `?

Reply:
No, this is correct.  The cast is to the type of the source `<Int16>` in this case because this value will be compared against the source.

Same on line 303